packages/ui/src/components/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import {
+  forwardRef,
+  HTMLChakraProps,
+  NumberDecrementStepper,
+  NumberIncrementStepper,
+  NumberInput as ChakraNumberInput,
+  NumberInputField,
+  NumberInputStepper,
+  UseNumberInputProps,
+} from '@chakra-ui/react';
+
+type NumberInputSize = 'sm' | 'md' | 'lg';
+
+type NumberInputMode = 'numeric' | 'decimal';
+
+export interface NumberInputProps extends Omit<HTMLChakraProps<'div'>, keyof UseNumberInputProps> {
+  /**
+   * Determines number input's paddings and height.
+   */
+  size: NumberInputSize;
+  /**
+   * Determines type of data that might be entered by the user (numeric or decimal).
+   */
+  inputMode?: NumberInputMode;
+  /**
+   * Determines if number input is disabled or not .
+   */
+  disabled?: boolean;
+  value?: number;
+  onChange?: (stringValue: string, numberValue: number) => void;
+  max?: number;
+  min?: number;
+  precision?: number;
+  step?: number;
+}
+
+export const NumberInput = forwardRef<NumberInputProps, 'input'>(
+  (
+    {
+      size,
+      inputMode = 'decimal',
+      disabled = false,
+      'aria-describedby': ariaDescribedby,
+      'aria-label': ariaLabel,
+      'aria-labelledby': ariaLabelledby,
+      ...props
+    },
+    ref,
+  ) => {
+    return (
+      <ChakraNumberInput
+        size={size}
+        inputMode={inputMode}
+        isDisabled={disabled}
+        aria-describedby={ariaDescribedby}
+        aria-label={ariaLabel}
+        aria-labelledby={ariaLabelledby}
+        ref={ref}
+        {...props}
+      >
```

Comment:
Type '{ children: Element[]; form?: string | undefined; pattern?: string | undefined; defaultValue?: string | number | readonly string[] | undefined; id?: string | undefined; onFocus?: FocusEventHandler<HTMLInputElement> | undefined; ... 659 more ...; ref: ForwardedRef<...>; }' is not assignable to type 'NumberInputProps'.
  Types of property 'defaultValue' are incompatible.
    Type 'string | number | readonly string[] | undefined' is not assignable to type 'StringOrNumber | undefined'.
      Type 'readonly string[]' is not assignable to type 'StringOrNumber | undefined'.
        Type 'readonly string[]' is not assignable to type 'string'.ts(2322)
(alias) const ChakraNumberInput: ComponentWithAs<"div", NumberInputProps>
import ChakraNumberInput
NumberInput

React component that provides context and logic to all number input sub-components.

![image](https://user-images.githubusercontent.com/59486546/125317417-48442a80-e339-11eb-958f-977a313fbd13.png)

Reply:
I cannot see the comment it relates to, but to make this error disappear add `defaultValue?: string | number;` to `NumberInputProps` interface